src/Adapter/Form/ChoiceProvider/TaxRuleGroupChoiceProvider.php

```diff
@@ -53,19 +82,26 @@ public function getChoices()
      */
     public function getChoicesAttributes(): array
     {
-        $attrs = [];
+        $address = new Address();
+        $address->id_country = $this->countryId;
+
+        $taxRates = [];
         foreach ($this->getRules() as $rule) {
-            // Keep first one found
-            if (!empty($attrs[$rule['name']]['data-tax-rate'])) {
-                continue;
+            $taxRulesGroupId = new TaxRulesGroupId((int) $rule['id_tax_rules_group']);
+            $stateId = $this->taxRulesGroupRepository->getTaxRulesGroupDefaultStateId($taxRulesGroupId, new CountryId($this->countryId));
+            if (!$stateId) {
+                $taxRate = $this->taxComputer->getTaxRate($taxRulesGroupId, new CountryId($this->countryId));
+            } else {
+                $taxRate = $this->taxComputer->getTaxRateByState($taxRulesGroupId, new CountryId($this->countryId), new StateId($stateId));
```

Comment:
```suggestion
            if (!$stateId) {
                $taxRate = $this->taxComputer->getTaxRate($taxRulesGroupId, new CountryId($this->countryId));
                $stateIsoCode = '';
            } else {
                $taxRate = $this->taxComputer->getTaxRateByState($taxRulesGroupId, new CountryId($this->countryId), new StateId($stateId));
                $state = new State($stateId);
                $stateIsoCode = $state->iso_code;
```

This will prevent instantiating a State all the time even when state ID is zero (which is most of the time)